src/platform/macos/mod.rs

```diff
@@ -309,11 +310,13 @@ enum SendData<'a> {
     OutOfLine(Option<OsIpcSharedMemory>),
 }
 
-const MAX_INLINE_SIZE: usize = 45 * 1024 * 1024;
+lazy_static! {
+    static ref MAX_INLINE_SIZE: AtomicUsize = AtomicUsize::new(45 * 1024 * 1024);
```

Comment:
I'd rather just go with `usize::max_value()` for the starting value... Not least to make sure the auto-shrinking code actually gets exercised.

(On that note, to really test this, the test case would need to do a sequence of sends at sizes above and below 48 MiB...)